### PR TITLE
Swtich to using ComponentFactory.CreateClient

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/tests/BlobConfigurationTests.cs [hunk 1]
```diff
@@ -65,7 +65,7 @@ public async Task BlobClient_CanConnect_EndPoint()
                 {
                     cb.AddInMemoryCollection(new Dictionary<string, string>()
                     {
-                        {"CustomConnection:endpoint", account.Endpoint },
+                        {"CustomConnection:serviceUri", account.Endpoint },
```
Comment (kasobol-msft) [inline sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/tests/BlobConfigurationTests.cs:68]: should we call this out in changelogs? it's a break between previews isn't it?

File: sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Queues/src/QueueServiceClientProvider.cs [hunk 2]
```diff
@@ -43,28 +43,20 @@ internal class QueueServiceClientProvider : StorageClientProvider<QueueServiceCl
...
+        protected override QueueServiceClient CreateClient(IConfiguration configuration, TokenCredential tokenCredential, QueueClientOptions options)
         {
             var originalEncoding = options.MessageEncoding;
             options.MessageEncoding = QueueMessageEncoding.None;
-            var nonEncodingClient = new QueueServiceClient(connectionString, options);
+            var nonEncodingClient = base.CreateClient(configuration, tokenCredential, options);
             options.MessageDecodingFailed += CreateMessageDecodingFailedHandler(nonEncodingClient);
             options.MessageEncoding = originalEncoding;
-            return new QueueServiceClient(connectionString, options);
-        }
```
Comment (kasobol-msft) [inline sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Queues/src/QueueServiceClientProvider.cs:58]: how is this going to work if connection string is used? I.e. how all these encoding related things are going to hookup now in that case?

Comment (pakrym) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Queues/src/QueueServiceClientProvider.cs:58]: Same as before. `IConfiguration configuration` can represent both a connection string or a section so this method is a replacement for the two we had before.

This line would create a client using the connection string or the serviceUri but without the encoding.

```
            var nonEncodingClient = base.CreateClient(configuration, tokenCredential, options);

```
and this one would create the main one:

```
            return base.CreateClient(configuration, tokenCredential, options);
```

File: sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/tests/BlobConfigurationTests.cs (hunk 1 above)
Comment (pakrym) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/tests/BlobConfigurationTests.cs:68]: Yes, good point.

**Review by kasobol-msft — APPROVED** [review]

**Pull request merged**